### PR TITLE
[AutoDiff] Fix Missed Adjoint -> Pullback renamings

### PR DESCRIPTION
**Pull request by bartchr808**
Instead of referring to the pullback function as the adjoint in the compiler code, refer to it as the pullback. This PR addresses some missed name changes.

### CODE REVIEW

**Review by rxwei — APPROVED** [review]

**Comment by bartchr808** [comment]
@swift-ci please test tensorflow

File: lib/SILOptimizer/Mandatory/Differentiation.cpp [hunk 1]
```diff
@@ -4492,27 +4492,29 @@ class PullbackEmitter final : public SILInstructionVisitor<PullbackEmitter> {
...
           // Allocate and zero initialize a new local buffer using
           // `getAdjointBuffer`.
-          builder.setInsertionPoint(adjoint.getEntryBlock());
+          builder.setInsertionPoint(pullback.getEntryBlock());
           getAdjointBuffer(origBB, activeValue);
         } else {
-          // Create and register adjoint block argument for the active value.
-          auto *adjointArg = adjointBB->createPhiArgument(
+          // Create and register pullback block argument for the active value.
+          auto *adjointArg = pullbackBB->createPhiArgument(
```
Comment (dan-zheng) [inline lib/SILOptimizer/Mandatory/Differentiation.cpp]: `adjointArg` -> `pullbackArg`

File: lib/SILOptimizer/Mandatory/Differentiation.cpp [hunk 2]
```diff
@@ -4492,27 +4492,29 @@ class PullbackEmitter final : public SILInstructionVisitor<PullbackEmitter> {
...
               getRemappedTangentType(activeValue->getType()),
               ValueOwnershipKind::Guaranteed);
           activeValueAdjointBBArgumentMap[{origBB, activeValue}] = adjointArg;
```
Comment (dan-zheng) [inline lib/SILOptimizer/Mandatory/Differentiation.cpp]: `activeValueAdjointBBArgumentMap` -> `activeValuePullbackBBArgumentMap`

File: lib/SILOptimizer/Mandatory/Differentiation.cpp [hunk 3]
```diff
@@ -4492,27 +4492,29 @@ class PullbackEmitter final : public SILInstructionVisitor<PullbackEmitter> {
...
         }
       }
       // Add a pullback struct argument.
-      auto *pbStructArg = adjointBB->createPhiArgument(
+      auto *pbStructArg = pullbackBB->createPhiArgument(
           pbStructLoweredType, ValueOwnershipKind::Guaranteed);
       pullbackStructArguments[origBB] = pbStructArg;
-      // - Create adjoint trampoline blocks for each successor block of the
+      // - Create pullback trampoline blocks for each successor block of the
       //   original block. Adjoint trampoline blocks only have a pullback
```
Comment (dan-zheng) [inline lib/SILOptimizer/Mandatory/Differentiation.cpp]: Rename occurrences of `adjoint` to `pullback`.
Also fix last clause: `trampoline adjoint values` -> `passing adjoint values`.

```
      //   original block. Pullback trampoline blocks only have a pullback
      //   struct argument, and branch from the pullback successor block to the
      //   pullback original block, passing adjoint values of active values.
```

File: lib/SILOptimizer/Mandatory/Differentiation.cpp [hunk 4]
```diff
@@ -4666,11 +4668,11 @@ class PullbackEmitter final : public SILInstructionVisitor<PullbackEmitter> {
...
-                  materializeAdjointDirect(activeValueAdj, adjLoc);
+                  materializeAdjointDirect(activeValueAdj, pbLoc);
               // Emit cleanups for children.
               if (auto *cleanup = concreteActiveValueAdj.getCleanup()) {
                 cleanup->disable();
-                cleanup->applyRecursively(builder, adjLoc);
+                cleanup->applyRecursively(builder, pbLoc);
               }
               trampolineArguments.push_back(concreteActiveValueAdj);
               // If the adjoint block does not yet have a registered adjoint
```
Comment (dan-zheng) [inline lib/SILOptimizer/Mandatory/Differentiation.cpp]: `adjoint block` -> `pullback block`

**Comment by dan-zheng** [comment]
I left some comments with more `adjoint` -> `pullback` renamings.
Don't worry about them, I'll handle them quickly!

**Pull request merged**